coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,300 @@
+        # line is 80.
+
+        print('The length of this line is 38')
+        """
+        example_langague = 'python'
+        importance_reason = """
+        Too long lines make your code very difficult to read and maintain.
+        """
+        fix_suggestions = """
+        Splitting long lines of code into multiple shorter lines whenever
+        possible. Avoiding the usage of inline language specific constructs
+        whenever they results in too long lines.
+        """
+    max_line_length = Taste[int](
+        'Maximum number of character for a line.',
+        (79, 80), default=80)
+
+
+@Length.subaspect
+class FileLength:
+    """
+    This aspect checks for files with a large number of lines in your codebase.
+    """
+    class docs:
+        example = """
+        # This file would be a large file if we assume that the max number of
+        # lines per file is 10
+
+        class node:
+            def __init__(self, value, leftmostchild, leftsibling):
+                self.value=value
+                self.leftmostchild=leftmostchild
+                self.leftsibling=leftsibling
+
+        # This is example is just showing what this aspect is about, because the
+        # max number of lines per file is usually 999.
+        """
+        example_language = 'python'
+        importance_reason = """
+        Too long programs (or files) are difficult to read, maintain and
+        understand.
+        """
+        fix_suggestions = """
+        Spliting files into modules, writing shorter methods and classes.
+        """
+    max_file_length = Taste[int](
+        'Maximum number of line for a file',
+        (999,), default=999)
+
+
+@Formatting.subaspect
+class Spacing:
+    """
+    This aspect checks on the spacing in your codebase (trailing
+    spaces, empty lines, indentation etc...)
+    """
+    class docs:
+        example = """
+        # Here is an example of code with spacing issues including trailing
+        # spaces, optional/useless blank lines and missing space aroud
+        # operators.
+
+
+
+        def func(   ):
+            return        37*-+2
+        """
+        example_language = 'python'
+        importance_reason = """
+        Useless spacing affects the readability and maintainability of your
+        code.
+        """
+        fix_suggestions = """
+        Removing the trailing spaces and the meaningless blank lines.
+        """
+
+
+@Spacing.subaspect
+class TrailingSpace:
+    """
+    This aspect checks for trailing spaces in your code.
+    """
+    class docs:
+        example = """
+        # Here is an example of code with trailing spaces.
+
+        def func(  a   ):
+            pass
+        """
+        example_language = 'python'
+        importance_reason = """
+        Trailing spaces make code less readablen and maintainaible.
+        """
+        fix_suggestions = """
+        Removing the trailing spaces.
+        """
+    allow_traling_spaces = Taste[bool](
```

Comment:
I wonder if this should be a taste or if we can remove it, if the user uses that aspect he'll want to check forthat so why make a setting? Also typo...